### PR TITLE
Port from testium-core

### PR DESCRIPTION
**Pull request by jkrems**
Copied the `lib/processes/proxy/cookies.js` file and added some tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# testium-cookie
+
+WebDriver does not return status codes or response headers.
+This module stores those in a special cookie that can be read
+from normal WebDriver methods.
+
+## modifyResponse(res)
+
+Modifies and existing node http response to inject the `_testium_` cookie.
```
Comment (khoomeister) [inline README.md]: s/and/an?

Comment (jkrems) [inline README.md]: Thanks! Fixed.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+This module stores those in a special cookie that can be read
+from normal WebDriver methods.
+
+## modifyResponse(res)
+
+Modifies an existing node http response to inject the `_testium_` cookie.
+
+## parseTestiumCookie(cookies)
+
+Takes either a cookie dictionary or a list of cookie objects and extracts the `_testium` meta data. It returns an object with a `headers` and a `statusCode` property.
```
Comment (khoomeister) [inline README.md]: s/_testium/_testium_/ ?

Comment (jkrems) [inline README.md]: Fixed as well.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+WebDriver does not return status codes or response headers.
+This module stores those in a special cookie that can be read
+from normal WebDriver methods.
+
+## modifyResponse(res)
+
+Modifies an existing node http response to inject the `_testium_` cookie.
+
+## parseTestiumCookie(cookies)
```
Comment (khoomeister) [inline README.md]: Should this be `getTestiumCookie`?  Additionally, I don't see `parseTestiumCookie` being exported.

Comment (jkrems) [inline README.md]: Yep, copy/paste error. Fixed.

File: test/modify-response.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  t.equal(res.headers['cache-control'], 'no-store',
+    'it overrides the cache-control header');
+
+  t.equal(res.headers['content-type'], 'text/html',
+    'it keeps other headers around');
+
+  t.equal(res.headers['set-cookie'], 'user=robin; path=/',
+    'it preserves the original set-cookie header');
+
```
Comment (khoomeister) [inline test/modify-response.js]: An additional check could be for `200` status code in actual response.

Comment (khoomeister) [inline test/modify-response.js]: And additional check for response.statusCode < 400 (i.e. no adjustment to status code)

Comment (jkrems) [inline test/modify-response.js]: Actually, `201`. Good point.

File: testium-cookie.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  return encodedData;
+}
+exports.encodeMetaData = encodeMetaData;
+
+function buildCookie(headers, statusCode) {
+  var jsonData = JSON.stringify({
+    headers: headers,
+    statusCode: statusCode
+  });
+  var encodedData = encode(jsonData);
```
Comment (khoomeister) [inline testium-cookie.js]: It looks like `encodedData` and `jsonData` variables aren't required since its purpose is fulfilled by `encodeMetaData`?

Comment (jkrems) [inline testium-cookie.js]: Added jshint to catch this in `npm test`.

**Comment by khoomeister** [comment]
:shipit:

**Pull request merged**